### PR TITLE
Fix the git release tag matching for CI/CD.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

Fix the git release tag matching for continuous deployment.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ parameters:
     default: "owner/repo"
   git_release_tag:
     type: string
-    default: v10\.[0-9]*\.[0-9]*$
+    default: v[0-9]+\.[0-9]+\.[0-9]+[\S]*$
```
Comment (alexander-kulikovskii) [inline .circleci/config.yml:14]: nice fix
maybe add only allowed words?
because such tag also will work `v10.4.4-_+beta..rc-123`

Comment (pengdev) [reply .circleci/config.yml:14]: I think we shouldn't limit the string after the version name, as we may want to introduce other variants of version name, e.g. alpha. 
And matching the initial part of the version name also matches the previous behavior.

Comment (yunikkk) [reply .circleci/config.yml:14]: Do we need to add regex for the starting part `v10.` ? It'll match `v0.` for example.

File: .circleci/config.yml [hunk 2]
```diff
@@ -208,7 +208,7 @@ commands:
       - run:
           name: Start internal CI release flow
           command: |
-            if [[ $CIRCLE_TAG == << pipeline.parameters.git_release_tag >> ]]; then
+            if [[ $CIRCLE_TAG =~ << pipeline.parameters.git_release_tag >> ]]; then
```
Comment (yunikkk) [inline .circleci/config.yml:211]: Hmm, but how did it work previously? We've had regex for quite a long time it seems

Comment (yunikkk) [reply .circleci/config.yml:211]: Ah, or it had worked with simple wildcard but when we've switched to the full regexps it stopped right

File: .circleci/config.yml (hunk 1 above)
Comment (pengdev) [reply .circleci/config.yml:14]: I guess it's our responsibility to cut the release with correct version name? I think the expression here is primarily used for distinguishing the release tags from the other tags we might create?

File: .circleci/config.yml (hunk 2 above)
Comment (pengdev) [reply .circleci/config.yml:211]: Need to run some real-word test though.. probably try it out in the next beta release.

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by Chaoba — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (yunikkk) [reply .circleci/config.yml:14]: Yeah, however why allow the incorrect and broad regex patterns? E.g. would `v10\.[0-9]+\.[0-9]+[\S]*$` be more strict and better?

Comment (pengdev) [reply .circleci/config.yml:14]: We will eventually have version 11 and so on, it's easy to miss the ci config when releasing new versions?

Comment (yunikkk) [reply .circleci/config.yml:14]: Hmm, from this point of view... Not sure that we won't change CI on the major release update, but ok, on with leaving as is

**Review by yunikkk — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Pull request merged**